src/include/crowbar/complex.rb

```diff
@@ -829,126 +807,30 @@ def HandleAskOnError(key, event)
-            # text entry label
-            InputField(Id(:username), Opt(:hstretch), _("User Name")),
-            # text entry label
-            Password(Id(:pw1), Opt(:hstretch), _("Password")),
-            # text entry label
-            Password(Id(:pw2), Opt(:hstretch), _("Repeat the Password")),
-            VSpacing(0.5),
-            ButtonBox(
-              PushButton(Id(:ok), Label.OKButton),
-              PushButton(Id(:cancel), Label.CancelButton)
-            ),
-            VSpacing(0.5)
-          ),
-          HSpacing(1)
-        )
-      )
-
-      if key == "edit_user"
-        UI.ChangeWidget(Id(:username), :Value, @current_user)
-        UI.ChangeWidget(Id(:pw1), :Value, @users[@current_user]["password"] || "")
-        UI.ChangeWidget(Id(:pw2), :Value, @users[@current_user]["password"] || "")
-      end
-
-      ret = :not_next
-      name = ""
-
-      while true
-        ret = UI.UserInput
-        break if ret == :cancel
-        if ret == :ok
-          name = UI.QueryWidget(Id(:username), :Value)
-          pass = UI.QueryWidget(Id(:pw1), :Value)
-
-          if name.empty?
-            # error popup
-            Popup.Error(_("User name cannot be empty."))
-            UI.SetFocus(Id(:username))
-            next
-          end
-
-          if pass != UI.QueryWidget(Id(:pw2), :Value)
-            # error popup
-            Popup.Error(_("The passwords do not match.\nTry again."))
-            UI.SetFocus(Id(:pw1))
-            next
-          end
-          if @users.key?(name) && (key == "add_user" || name != @current_user)
-            # error popup
-            Popup.Error(
-              Builtins.sformat(
-                _("User '%1' already exists.\nChoose a different name."),
-                name
-              )
-            )
-            UI.SetFocus(Id(:username))
-            next
-          end
-          @users.delete (@current_user) if key == "edit_user"
-          @users[name] = { "password" => pass }
-          break
-        end
-      end
-      UI.CloseDialog
-
-      if ret == :ok
-        @current_user = name
-        InitUsersTable("users_table")
-        UI.ChangeWidget(Id("delete_user"), :Enabled, @users.size > 0)
-        UI.ChangeWidget(Id("edit_user"), :Enabled, @users.size > 0)
-      end
+    def StoreAdminUser(key, event)
+      @admin_user = UI.QueryWidget(Id(key), :Value)
       nil
     end
 
-    # handler for user selection table
-    def HandleUsersTable(key, event)
-      selected = UI.QueryWidget(Id(key), :Value)
-      if selected != nil && selected != @current_user
-        @current_user = (selected == nil ? "" : selected)
-      end
-      if key == event["ID"] && event["EventReason"] == "Activated"
-        HandleAddEditUserButton("edit_user", { "ID" => "edit_user" })
-      end
+    def InitAdminPassword(id)
+      UI.ChangeWidget(Id(id), :Value, @admin_password)
       nil
     end
 
+    def ValidateAdminUser(key, event)
+      UI.QueryWidget(Id(key), :Value) != ""
```

Comment:
well, usually it is good practice to show popup with message what is wrong when validation failed. Something like
```
def ValidateAdminUser(key, event)
  return true if UI.QueryWidget(Id(key), :Value) != ""

  Yast::Popup.Error(_("Admin User cannot be empty"))
  false
end
```

Reply:
@jreidinger I used `validate_help` on the widget instead (and yes, it works :-))